main.cpp

```diff
@@ -231,214 +333,320 @@ struct BasketballCourt
+    float sellTickets(float ticketPrice);
+
     void scheduleNewGameDay(EventsSchedule newGameDay);
     EventsSchedule scheduledGameDay;
 };
 
-/*
-Thing 4) cafeteria
+void BasketballCourt::EventsSchedule::cancelAllMondayEvents(int numberOfMondayEvents)
+{
+    std::cout << numberOfMondayEvents << "events have been cancelled";
+}
+
+void BasketballCourt::EventsSchedule::rescheduleAllWednesdayEvents(int numberOfWednesdayEvents)
+{
+    std::cout << numberOfWednesdayEvents << "events have been rescheduled";
+}
+
+void BasketballCourt::EventsSchedule::organizeNextMonthsSchedule()
+{
+    int numberOfNextMonthsEvents = 18;
+    std::cout << "organize" << numberOfNextMonthsEvents << "next month";
+}
+
+void BasketballCourt::hostBasketballGames(EventsSchedule gameDay)
+{
+    gameDay.rescheduleAllWednesdayEvents(2);
+}
+
+void BasketballCourt::hostSportEvents()
+{
+    std::string nameOfEvent = "event name";
+    std::string eventDate = "date";
+    std::cout << nameOfEvent << "will take place on" << eventDate;
+}
+
+float BasketballCourt::sellTickets(float ticketPrice)
+{
+    int numberOfTicketsSold = 651.0;
+    float ticketsFloat;
+    ticketsFloat = float(numberOfTicketsSold);
+    float earnings = ticketPrice * ticketsFloat;
+    return earnings;
+}
+
+void BasketballCourt::scheduleNewGameDay(EventsSchedule newGameDay)
+{
+    newGameDay.organizeNextMonthsSchedule();
+    scheduledGameDay = newGameDay;
+}
 
- */
 struct Cafeteria
 {
-    //5 properties:
-    //    1) price of coffee (float)
     float priceOfCoffee = 3.2f;
-    //    2) amount of emplyees (int)
     int amountOfEmployees = 8;
-    //    3) price of sandwich (float)
     float priceOfSandwich = 4.8f;
-    //    4) amount of rent per month (float)
     float amountOfRentPerMonth = 1028.12f;
-    //    5) number of customers per month (int)
     int numberOfCustomersPerMonth = 964;
-    //3 things it can do:
-    //    1) sell coffee
-    float sellCoffee(float coffeePrice); //returns the amount of money earned
-    //    2) provide a place for friendly meetings
-    void providePlaceForFriendlyMeetings();
-    //    3) sell snacks
-    float sellSnacks(float snacksPrice); //returns the amount of money earned
 
+    float sellCoffee(float coffeePrice);
+    void providePlaceForFriendlyMeetings();
+    float sellSnacks(float snacksPrice);
 };
 
-/*
-Thing 5) audio department
+float Cafeteria::sellCoffee(float coffeePrice)
+{
+    int numberOfCoffeesSold = 132;
+    float coffeesFloat;
+    coffeesFloat = float(numberOfCoffeesSold);
+    float profit = coffeePrice * coffeesFloat;
+    return profit;
+}
+
+void Cafeteria::providePlaceForFriendlyMeetings()
+{
+    int time = 8;
+    std::string day = "Friday";
+    std::cout << "the meeting will take place on" << day << "at" << time << "o'clock";
+}
+
+float Cafeteria::sellSnacks(float snacksPrice)
+{
+    int numberOfSnacksSold = 44;
+    float snacksFloat;
+    snacksFloat = float(numberOfSnacksSold);
```

Comment:
Casting has not been covered, and this is a C-Style cast.
remove it
find another way to solve the warnings related to this type conversion.
- [ ] revise